app/services/update_course_stats.rb

```diff
@@ -64,7 +64,7 @@ def update_article_status
   end
 
   def update_average_pageviews
-    AverageViewsImporter.update_outdated_average_views(@course.articles)
+    AverageViewsImporter.update_outdated_average_views(@course.articles, update_service: self)
```

Comment:
I suggest starting with just one case (either this one, or the first branch of `Category#title_list_from_wiki` as I outlined above).

Reply:
Okay, noted